File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) 
    {
        //use counting sort with array value of int[101]
        int[] frequency = new int[101];
        
        //count frequency of each number
        for(int j : nums)
            frequency[j]++;
       
        //sum the previous to current one to get the number of elements before 
        //it in a sorted array 
        for(int i = 1 ; i < frequency.length ;i++)
            frequency[i] = frequency[i] + frequency[i-1];
        
        int[] solution = new int[nums.length];
        
        for(int j = 0 ; j < nums.length ; j++)
        {
            //cant use --nums[j] because it will be negative
            if(nums[j] == 0)
                solution[j] = 0;
            else
                solution[j] = frequency[nums[j]-1];
        }
        return solution;
    }
    
}

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        all_nums = [0]*101
        for i in nums:
            all_nums[i] +=1
        for i in range(1,101):
            all_nums[i] = all_nums[i]+all_nums[i-1]
        solution = [0]*len(nums)
        
        for i in range(len(nums)):
            if nums[i] == 0:solution[i] = 0
            else:
                solution[i] = all_nums[nums[i]-1]
        return solution